### PR TITLE
feat(sdk): revamp CreateQuestion and create question behaviour

### LINKED ISSUE #50082: Change CreateQuestion's API, implementation and improve default behaviour on saving questions
We should replace the `<CreateQuestion />` component's API, underlying implementation and docs. It should now consume the `<InteractiveQuestion />` directly and expose the same props. We should also improve [the behaviour after the question is being saved](https://www.loom.com/share/aafc92798b814f56bd989765153920dc).

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50082

### Description

This PR updates the underlying implementation of the `CreateQuestion` component to use `InteractiveQuestion` directly, and alters the default behaviour for saving questions.

### How to verify

- Use the `<CreateQuestion />` component
- There should be the "Show Visualization / Show Editor" button instead of tabs
- When saving the question, the question title should update. It should NOT show "Question not found"
- If you make a change to the saved question and click "Save" again, it should have the "- Modified" suffix

### Demo

[Private Loom](https://loom.com/share/aafc92798b814f56bd989765153920dc)

How it looks in Shoppy

CleanShot 2567-11-20 at 02 58 41@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - E2E

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SaveButton.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SaveButton.tsx:22]: This is based on [Maz's feedback in this private Loom](https://www.loom.com/share/aafc92798b814f56bd989765153920dc):

> If you're looking at a saved, unmodified question, I don't think the Save button should be visible at all, until/unless the query or visualization is modified. This applies to saving questions in general. I can't really think of a situation where we'd want to show the Save button but not have it be enabled.

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:86]: We used to call `loadQuestion` here which actually just reloads the original question, which doesn't match the expectation from the main Metabase app. This instead gets the created question from the save handler, passes it to `onSave` and replaces the current question / originalQuestion with the newly created question.

This allows the save modal to show the correct question's title, as well as adding the "- Modified" suffix when re-saving the question to a new question. Refer to the e2e test for the expected title change behavior.

File: e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts
Comment (heypoom) [inline e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts:47]: I got a feedback from Maz that we should probably show the actual model name or table name instead of "New question", but let's do that in a separate PR.

File: enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx:138]: Let's remove this component before GA in the next releases.

File: enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx:15]: Same prop as `InteractiveQuestion`, but without `questionId` and `children`. All props are optional.

File: enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx:27]: The save button is hidden when the user passes `isSaveEnabled={false}`, mostly useful for demonstration purposes (e.g. allowing to create a question to query data, but not to save them).

File: enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx:43]: Oisin is cooking a [better default layout](https://www.figma.com/proto/Vkrs53Rit3EPmCaIkI0ZBf/SDK-Default-Layout?page-id=96%3A9544&node-id=96-9565&node-type=frame&t=RTgs7uSw5DfUuUK3-0&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=96%3A11359&show-proto-sidebar=1) right now. I'll likely make a follow-up PR to reuse that layout here.

File: enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts:45]: Needed to provide the sensible defaults after saving a question.

File: frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx
Comment (heypoom) [inline frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx:29]: In dark-themed apps, the header looks ugly without this.

CleanShot 2567-11-20 at 00 34 21@2x

File: e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts
Comment (oisincoveney) [reply e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts:47]: I've made a utility function called `getQuestionTitle` in the `QuestionTitle` component (we can split that out) which gets the question name, adhoc description, or `New question` by default. We could probably extend that to be a `getEntityName` type of thing

**Review by oisincoveney — APPROVED** [review]
Looks good! I'll need to remind myself to make sure this component works before we merge the default layout.

File: e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts
Comment (heypoom) [reply e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts:47]: @oisincoveney Yep, I fiddled with `getQuestionTitle` a little bit yesterday but couldn't find an obvious method in the `Question` object that spits out the model or table name. I'll check with the querying folks

File: enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/public/CreateQuestion/CreateQuestion.tsx:68]: @oisincoveney tbh I didn't knew that `Group` existed (I usually use `Stack` with `Flex`), looks like Group is a 
 horizontal flex container 👀

**Comment by heypoom** [comment]
Merging this first to unblock the [Metabase 51 Release Webinar](https://metaboat.slack.com/archives/C063Q3F1HPF/p1732110044487209). I'll address any follow-up change requests in a subsequent PR.

**Pull request merged**
